serverless.yaml

```diff
@@ -79,11 +79,19 @@ provider:
       Resource:
         - !GetAtt FHIRBinaryBucket.Arn
         - !Join ['', [!GetAtt FHIRBinaryBucket.Arn, '/*']]
+    - Action:
+        - 'xray:PutTraceSegments'
+        - 'xray:PutTelemetryRecords'
+      Effect: Allow
+      Resource:
+        - '*'
   variableSyntax: "\\${((?!AWS)[ ~:a-zA-Z0-9._@'\",\\-\\/\\(\\)]+?)}" # Use this for allowing CloudFormation Pseudo-Parameters in your serverless.yml
   logs:
     restApi:
       executionLogging: true
       fullExecutionData: true
+  tracing:
+    apiGateway: true

```

Comment:
In this [blog post](https://www.serverless.com/blog/framework-release-v141) it says "If you’re using X-Ray Tracing for API Gateway you might want to enable it for your Lambda functions as well:". Should we enable lambda tracing as well? What's the difference between that and just API Gateway tracing?

Also "Due to CloudFormation limitations it's not possible to enable AWS X-Ray Tracing on existing deployments which don’t use tracing right now.

Please remove the old API Gateway and re-deploy it with tracing enabled if you want to use AWS X-Ray Tracing for API Gateway."

A headsup, that we might have to do some manual steps on the test env in the `chamonix-common` account, to allow the github workflow to successfully deploy to the test env.

Reply:
I think that only enabling tracing for APIGW is the way to go. 

The difference is that with this setup there won't be xray traces if you invoke the lambda function directly. In our case this is not an issue since the lambda function is always invoked by APIGW. APIGW has a native integration with Lambda, and enabling xray on APIGW makes the downstream Lambda calls also record xray traces

Also, I don't think Cloudformation deployment is an issue. I deployed it on my AWS account (as an update to my old stack) and it went from xray disabled to xray enabled just fine. I'd guess that the blog post is outdated. Serverless v1.41 came out in 2019-04-23. 
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html#cfn-apigateway-stage-tracingenabled